src/kafka-util/src/client.rs

```diff
@@ -160,9 +160,13 @@ impl ClientContext for MzClientContext {
         match level {
             Emerg | Alert | Critical | Error => {
                 self.record_error(log_message);
-                error!(target: "librdkafka", "{} {}", fac, log_message);
+                // We downgrade error messages to `warn!` level to avoid
+                // sending the errors to Sentry. Most errors are customer
+                // configuration problems that are not appropriate to send to
+                // Sentry.
+                warn!(target: "librdkafka", "error: {} {}", fac, log_message);
```

Comment:
alternatively we can just add a per-layer filter to the sentry layer to filter librdkafka?

Reply:
I thought about that! But I like the rule we have currently where error! logs are 1:1 with what gets reported to Sentry. And our log level guidance says:

> `ERROR`: Reports an error that certainly indicates data corruption, data loss, unavailability, or an invariant violation that results in undefined behavior.

So by that guidance we shouldn't be using the `error!` log level, as these logs do not _certainly_ indicate any of the above.

Reply:
ah yes, sounds good!